### PR TITLE
feat: Event configuration

### PR DESCRIPTION
**Pull request by yassine04e**
Copied from #85 

Review the project and add an event each time a participant is added or document is uploaded, ect.

Acceptance criteria : 
1) Each time a travel is created, an event needs to be registered, and it needs to stipulate that it is the start of the journey
2) Each time an activity is saved, an event needs to be registered
3) Each time a participant joins, an event needs to be registered
4) The icon needs to be updated and the screen needs to be inambiguous. So when no events are registered, clearly stipulate it and let the user the ability to go backl
5) [Optional] Each time a document is uploaded, an event is registered

### CODE REVIEW

File: app/src/main/java/com/github/se/travelpouch/model/events/Repository.kt [hunk 1]
```diff
@@ -22,18 +24,22 @@ interface EventRepository {
...
-   * This function adds an event to the collection of events in Firebase.
-   *
-   * @param event (Event) : the event we want to add on Firebase
-   * @param onSuccess (() -> Unit) : the function called when the event is correctly added to the
-   *   database
-   * @param onFailure ((Exception) -> Unit) : the function called when an error occurs during the
-   *   adding an event to the database
-   */
-  fun addEvent(event: Event, onSuccess: () -> Unit, onFailure: (Exception) -> Unit)
+  fun getNewDocumentReference(): DocumentReference
```
Comment (RemIsMyWaifuu) [inline app/src/main/java/com/github/se/travelpouch/model/events/Repository.kt:29]: Don't forget to document functions

File: app/src/test/java/com/github/se/travelpouch/model/events/EventRepositoryUnitTest.kt [hunk 2]
```diff
@@ -106,19 +121,20 @@ class EventRepositoryUnitTest {
...
+  //
+  //    // This test verifies that when we add a new event, the Firestore `collection()` method is
+  //    // called.
+  //    eventRepositoryFirestore.addEvent(event, onSuccess = {}, onFailure = {})
+  //
+  //    shadowOf(Looper.getMainLooper()).idle()
+  //
+  //    // Ensure Firestore collection method was called to reference the "events" collection
+  //    verify(mockDocumentReference).set(any())
+  //  }
```
Comment (RemIsMyWaifuu) [inline app/src/test/java/com/github/se/travelpouch/model/events/EventRepositoryUnitTest.kt:137]: Please remove these lines

File: app/src/test/java/com/github/se/travelpouch/model/activity/ActivityRepositoryUnitTest.kt [hunk 3]
```diff
@@ -113,16 +119,49 @@ class ActivityRepositoryUnitTest {
...
+    // success
+    //
+    //    // This test verifies that when we add a new event, the Firestore `collection()` method is
+    //    // called.
+    //    activityRepositoryFirestore.addActivity(activity, onSuccess = {}, onFailure = {})
+    //
+    //    shadowOf(Looper.getMainLooper()).idle()
+    //
+    //    // Ensure Firestore collection method was called to reference the "events" collection
+    //    verify(mockDocumentReference).set(any())
```
Comment (RemIsMyWaifuu) [inline app/src/test/java/com/github/se/travelpouch/model/activity/ActivityRepositoryUnitTest.kt:164]: Please remove these lines if they are not used.

File: app/src/main/java/com/github/se/travelpouch/model/events/EventViewModel.kt [hunk 4]
```diff
@@ -29,21 +30,25 @@ class EventViewModel @Inject constructor(private val repository: EventRepository
...
-    repository.addEvent(event = event, onSuccess = { getEvents() }, onFailure = {})
-  }
+  //  /**
+  //   * This function adds a new event in Firebase
+  //   *
+  //   * @param event (Event) : a new event to add in Firebase
+  //   */
+  //  fun addEvent(event: Event) {
+  //    repository.addEvent(event = event, onSuccess = { getEvents() }, onFailure = {})
+  //  }
```
Comment (RemIsMyWaifuu) [inline app/src/main/java/com/github/se/travelpouch/model/events/EventViewModel.kt:53]: What is the purpose of this comment?

File: app/src/main/java/com/github/se/travelpouch/model/events/EventRepository.kt [hunk 5]
```diff
@@ -53,27 +68,29 @@ class EventRepositoryFirebase(private val db: FirebaseFirestore) : EventReposito
...
+  // the
+  //   *   database
+  //   * @param onFailure ((Exception) -> Unit) : the function called when an error occurs during
+  // the
+  //   *   adding an event to the database
+  //   */
+  //  override fun addEvent(event: Event, onSuccess: () -> Unit, onFailure: (Exception) -> Unit) {
+  //    performFirestoreOperation(
+  //        db.collection(collectionPath).document(event.uid).set(event), onSuccess, onFailure)
+  //  }
```
Comment (RemIsMyWaifuu) [inline app/src/main/java/com/github/se/travelpouch/model/events/EventRepository.kt:93]: What is the purpose of these lines?

**Review by RemIsMyWaifuu — CHANGES_REQUESTED** [review]
The timeline works as expected, atleast from what I could test by hand. Many lines were changed although the behavioural change is not that large. The integration of events is successful in my opinion.
You made a good use of DocumentReference, I didn't know that existed.

There are a few remaining elements to improve and questions to be answered before merging.
- There are commented lines that need to be deleted if they are not used or comments.
- The documentation for repository and modelview methods that include an eventDocumentReference hasn't been modified to reflect this change
- What is the purpose of the last 2 parameters in an event? i.e. `uidParticipant `and `listUploadedDocuments` 
I always see them set to null, seemingly serving no purpose.

So to recap, commented lines, documenting modified methods and the last 2 event fields need to be addressed.

File: app/src/main/java/com/github/se/travelpouch/model/events/Repository.kt [hunk 6]
```diff
@@ -22,18 +24,22 @@ interface EventRepository {
...
+  //   *
+  //   * @param event (Event) : the event we want to add on Firebase
+  //   * @param onSuccess (() -> Unit) : the function called when the event is correctly added to
+  // the
+  //   *   database
+  //   * @param onFailure ((Exception) -> Unit) : the function called when an error occurs during
+  // the
+  //   *   adding an event to the database
+  //   */
+  //  fun addEvent(event: Event, onSuccess: () -> Unit, onFailure: (Exception) -> Unit)
```
[the remark this answers is not in the document]
Comment (yassine04e) [reply app/src/main/java/com/github/se/travelpouch/model/events/Repository.kt:42]: Adding an event is obsolete. Now, we just create a new document reference and directly add the event in the transaction of the function we aim. 
For example, if a new activity is created, I create a new documetnRefernce for an event and in the fucntion add of the ActivityRepository, I set the event and add it in the trasaction lambda.
Is it clearer ?

File: app/src/main/java/com/github/se/travelpouch/model/events/Repository.kt (hunk 1 above)
Comment (yassine04e) [reply app/src/main/java/com/github/se/travelpouch/model/events/Repository.kt:29]: Done

File: app/src/test/java/com/github/se/travelpouch/model/events/EventRepositoryUnitTest.kt (hunk 2 above)
Comment (yassine04e) [reply app/src/test/java/com/github/se/travelpouch/model/events/EventRepositoryUnitTest.kt:137]: Done

File: app/src/test/java/com/github/se/travelpouch/model/activity/ActivityRepositoryUnitTest.kt (hunk 3 above)
Comment (yassine04e) [reply app/src/test/java/com/github/se/travelpouch/model/activity/ActivityRepositoryUnitTest.kt:164]: Done

File: app/src/main/java/com/github/se/travelpouch/model/events/EventViewModel.kt (hunk 4 above)
Comment (yassine04e) [reply app/src/main/java/com/github/se/travelpouch/model/events/EventViewModel.kt:53]: No, it is useless. It has been removed

File: app/src/main/java/com/github/se/travelpouch/model/events/EventRepository.kt (hunk 5 above)
Comment (yassine04e) [reply app/src/main/java/com/github/se/travelpouch/model/events/EventRepository.kt:93]: Nothing. I tend to forget to do a cleanup. Sorry for the inconvenience

**Comment by yassine04e** [comment]
Ok, I added missing documentation. 
Regarding the fields of events, I think it is better if a clean it in another PR. Don't you think ?

**Comment by RemIsMyWaifuu** [comment]
> Ok, I added missing documentation. Regarding the fields of events, I think it is better if a clean it in another PR. Don't you think ?

??? humm ok

**Comment by yassine04e** [comment]
Ok, but this will add more lines
